server/etcdserver/corrupt.go

```diff
@@ -212,17 +212,6 @@ func (cm *corruptionChecker) PeriodicCheck() error {
 		}
 		checkedCount++
 
-		// leader expects follower's latest revision less than or equal to leader's
-		if p.resp.Header.Revision > rev2 {
```

Comment:
PR doesn't state why remove this check. It seems unrelated so it would be good to confirm.

Goal of this check is to detect cases where follower got corrupted in a way that would lead them to have higher revision number. In such cases follower might never respond to leader request for `HashByRev` follower might have already compacted this revision. To detect such cases leader verifies that follower current revision is lower than its own revision.

Why would you want to remove this check?

Reply:
Please note that if there are new K/V data during the process, then the follower's rev may bigger than the leader.

Reply:
Nope, please check order of the operations. We first request and collect hashes from peers with their latest revision, and only after that we again check local latest revision. 
Due to linearizability of etcd to consecutive requests should result in revision not decreasing.

Reply:
In theory, it's still possible for the follower's rev to be bigger than the leader, although it's not easy to reproduce the situation in real production environment. The reason is the peers communication do not go through raft.

Reply:
Let me take one step back. Just reverted the change, the only reason is that it's unrelated change.